server/src/com/cloud/user/AccountManagerImpl.java

```diff
@@ -677,6 +679,17 @@ public boolean deleteAccount(AccountVO account, long callerUserId, Account calle
         return cleanupAccount(account, callerUserId, caller);
     }
 
+    protected List<VolumeVO> getExpungedInstanceRootVolume(long instanceId) {
+        SearchBuilder<VolumeVO> sb = _volumeDao.createSearchBuilder();
+        sb.and("instanceId", sb.entity().getInstanceId(), SearchCriteria.Op.EQ);
+        sb.and("vType", sb.entity().getVolumeType(), SearchCriteria.Op.EQ);
+        sb.done();
+        SearchCriteria<VolumeVO> c = sb.create();
+        c.setParameters("instanceId", instanceId);
+        c.setParameters("vType", Volume.Type.ROOT);
+        return _volumeDao.customSearchIncludingRemoved(c, null);
```

Comment:
This method does not return expunged volume, but root volume of a VM. Could there be a case where a VM can have more than one root volume? If not, let's rename the method to say `getInstanceRootVolume` and return just the VolumeVO instead a list.